frontend/src/App.js

```diff
@@ -14,35 +14,37 @@ function App() {
         const fetchedRoutes = await fetchAllRoutes();
         setAllRoutes(fetchedRoutes);
       } catch (error) {
-        console.error("Failed to fetch routes:", error);
+        console.error('Failed to fetch routes:', error);
       }
     };
     fetch();
   }, []);
 
   useEffect(() => {
     if (selectedRoute) {
-      const route = allRoutes.find(route => route.routeId === parseInt(selectedRoute, 10));
+      const route = allRoutes.find((route) => route.routeId === parseInt(selectedRoute, 10));
       setRouteDetails(route);
+      console.log("Selected Route Details:", route);
     }
   }, [selectedRoute, allRoutes]);
 
   return (
     <div className="App">
       <h1>Route Details</h1>
-      <Dropdown 
+      <Dropdown
         routes={allRoutes}
         selectedRoute={selectedRoute}
         setSelectedRoute={setSelectedRoute}
         showId={false}
       />
-      {routeDetails && (
+      {routeDetails ? (
         <div>
-          <h2>{routeDetails.name}</h2>
+          <p>Name: {routeDetails.name}</p>
           <p>ID: {routeDetails.routeId}</p>
-          <p>Sequence: {routeDetails.sequence}</p>
-          <p>Object: {routeDetails.object}</p>
+          <p>Technical Level: {routeDetails.technicalLevelName || 'N/A'}</p>
         </div>
```

Comment:
Your logic to display is good but this isn't a reusable component is it?

Wrap this HTML up in a component file `components/ClusterCard.tsx` and then import it here.